flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala

```diff
@@ -119,4 +119,35 @@ class DataSetWindowAggregateITCase(
     val results = windowedTable.toDataSet[Row].collect()
     TestBaseUtils.compareResultAsText(results.asJava, expected)
   }
+
+  @Test
+  def testEventTimeSessionGroupWindow(): Unit = {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env, config)
+
+    val table = env.fromCollection(data).toTable(tEnv, 'long, 'int, 'string)
+    val windowedTable = table
+      .groupBy('string)
+      .window(Session withGap 7.milli on 'long as 'w)
+      .select('string, 'string.count, 'w.start, 'w.end)
+
+    val results = windowedTable.toDataSet[Row].collect()
+
+    val expected = "Hallo,1,1970-01-01 00:00:00.002,1970-01-01 00:00:00.009\nHello world,1," +
+      "1970-01-01 00:00:00.008,1970-01-01 00:00:00.015\nHello world,1," +
+      "1970-01-01 00:00:00.016,1970-01-01 00:00:00.023\nHello,3,1970-01-01 00:00:00.003," +
+      "1970-01-01 00:00:00.013\nHi,1,1970-01-01 00:00:00.001,1970-01-01 00:00:00.008"
+    TestBaseUtils.compareResultAsText(results.asJava, expected)
+  }
+
+  @Test(expected = classOf[UnsupportedOperationException])
+  def testAlldEventTimeSessionGroupWindow(): Unit = {
```

Comment:
Yes, I have discussed this with fabian, and we will add non-grouped session windows for batch tables in  [FLINK-5219](https://issues.apache.org/jira/browse/FLINK-5219).